src/idl_parser.cpp

```diff
@@ -3203,6 +3203,19 @@ CheckedError Parser::ParseRoot(const char *source, const char **include_paths,
   // Check that all types were defined.
   for (auto it = structs_.vec.begin(); it != structs_.vec.end();) {
     auto &struct_def = **it;
+    for (auto fld_it = struct_def.fields.vec.begin(); fld_it != struct_def.fields.vec.end(); ++fld_it) {
```

Comment:
I think it's cleaner to internalize the iteration within CheckPrivateLeak. I get that its hard to abstract over the different ways of iterating over structs and enums, but it might be worth giving up a little DRY in favor of encapsulation.

Reply:
But we aren't really iterating over the enums, we are just using the existing code to do so:

https://github.com/google/flatbuffers/blob/af9827c4495481d5e0fc942a6c82abbf55c09ca8/src/idl_parser.cpp#L3273&L3297

Reply:
True but I think it'd be clearer to not use the existing code to iterate over enum variants.

If that loop was in its own method (and imo it should have been), it would be called `CheckUnionTypesAreSupported`. That is a different check than whether checking whether type leaks its inner types.

Reply:
Okay, if that's the case I can also move it to the function. I will just need to see how the code would change.